src/node/interfaces.cpp

```diff
@@ -973,7 +973,10 @@ class MinerImpl : public Mining
         {
             WAIT_LOCK(notifications().m_tip_block_mutex, lock);
             notifications().m_tip_block_cv.wait_for(lock, timeout, [&]() EXCLUSIVE_LOCKS_REQUIRED(notifications().m_tip_block_mutex) {
-                return (notifications().m_tip_block != current_tip && notifications().m_tip_block != uint256::ZERO) || chainman().m_interrupt;
+                // Although C++17 allows safe comparison of optional<T> with
+                // another T, we need to wait for m_tip_block to be set AND
```

Comment:
In commit "Make m_tip_block an std::optional" (2fff10a656edb8bec7d45433bf431fd233d0fd81)

IMO would be less confusing to drop comment about c++17 and just say "We need to wait for m_tip_block to be set AND for the value to be different than the current_tip value" which explains why there are two conditions.